addons/payment_demo/models/payment_transaction.py

```diff
@@ -118,8 +120,11 @@ def _send_void_request(self, amount_to_void=None):
         if self.provider_code != 'demo':
             return child_void_tx
 
-        notification_data = {'reference': self.reference, 'simulated_state': 'cancel'}
-        self._handle_notification_data('demo', notification_data)
+        notification_data = {
+            'reference': (child_void_tx or self).reference,
+            'simulated_state': 'cancel'
```

Comment:
Nitpicking
```suggestion
            'simulated_state': 'cancel',
```
This way, if someone ever has to add a new line after, the diff will be a single line diff, instead of 2 =]